### PR TITLE
Add C++14 initializiers for lambda captures

### PR DESCRIPTION
**Pull request by mknejp**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -385,6 +386,35 @@ int three = identity(3); // == 3
...
+
+auto generator = [x = 0] () mutable { return x++; };
+auto a = generator(); // a = 0
+auto b = generator(); // b = 1
+```
+Because it is now possible to _move_ (or _forward_) values into a lambda that could previously be only captured by copy or reference we can now capture move-only types in a lambda by value. Note that in the below example the `p` in the capture-list of `task2` on the left-hand-side of `=` is a new variable private to the lambda body and does not refer to the original `p`.
+```c++
+auto p = std::make_unique<int>(1);
+
+auto task1 = [=] { return *p = 5; } // ERROR: std::unique_ptr cannot be copied
```
Comment (AnthonyCalandra) [inline README.md]: Forgot a semicolon here.

File: README.md [hunk 2]
```diff
@@ -385,6 +386,35 @@ int three = identity(3); // == 3
 std::string foo = identity("foo"); // == "foo"
 ```
 
+### Lambda capture initializers
+This allows creating lambda captures initialized with arbitrary expressions. The name given to the captured value does not need to be related to any variables in the enclosing scopes and introduces a new name inside the lambda body. The initializing expression is evaluated when the lambda is _created_ (not when it is _invoked_).
+```c++
+int factory(int i) { return i * 10; }
+auto f = [x = factory(2)] { return x; }; // returns 20
+
+auto generator = [x = 0] () mutable { return x++; };
```
Comment (AnthonyCalandra) [inline README.md]: Could we add a short explanation of what `mutable` does and why we need it here? Doesn't matter if it's even a sentence.

File: README.md (hunk 1 above)
Comment (AnthonyCalandra) [inline README.md]: Also, I think the return statement doesn't belong here.

Comment (AnthonyCalandra) [inline README.md]: Also, I think the return statement doesn't belong here.

File: README.md (hunk 2 above)
Comment (mknejp) [inline README.md]: Wouldn't that be part of the C++11 lambda introduction?

Comment (AnthonyCalandra) [inline README.md]: Yes, that would be right. Sorry I was slightly confusing when I used the word "here" (I was referring to why it's needed in the snippet).

**Comment by mknejp** [comment]
Added the feedback and also extended the C++11 lambda section with `mutable` since it wasn't mentioned.

**Comment by AnthonyCalandra** [comment]
Looks good to me. Thanks for contributing, Miro!

**Pull request merged**